File: apps/problems/binary-search/boilerplate-full/function.cpp

#include <iostream>
#include <vector>
#include <string>

##USER_CODE_HERE##

int main() {
  int size_arr;
  std::cin >> size_arr;
  std::vector<int> arr(size_arr);
  for(int i = 0; i < size_arr; ++i) std::cin >> arr[i];
  std::cin >> x;
  int result = binSearch(arr, x);
  std::cout << result << std::endl;
  return 0;
}
    

File: apps/problems/binary-search/boilerplate-full/function.rs

use std::io::{self, BufRead};

##USER_CODE_HERE##

fn main() {
  let stdin = io::stdin();
  let mut input = stdin.lock().lines().map(|l| l.unwrap());
  let size_arr: usize = input.next().unwrap().parse().unwrap();
let arr: Vec<i32> = input.take(size_arr).map(|s| s.parse().unwrap()).collect();
  let x: i32 = input.next().unwrap().parse().unwrap();
  let result = binSearch(arr, x);
  println!("{}", result);
}
    